dhcpv6/dhcpv6message.go

```diff
@@ -57,31 +64,20 @@ func NewSolicitWithCID(duid Duid, modifiers ...Modifier) (*Message, error) {
 	})
 	m.AddOption(oro)
 	m.AddOption(&OptElapsedTime{})
-	// FIXME use real values for IA_NA
-	iaNa := &OptIANA{}
-	iaNa.IaId = [4]byte{0xfa, 0xce, 0xb0, 0x0c}
-	iaNa.T1 = 0xe10
-	iaNa.T2 = 0x1518
-	m.AddOption(iaNa)
+	if len(hwaddr) < 4 {
+		return nil, errors.New("short hardware addrss: less than 4 bytes")
+	}
+	l := len(hwaddr)
+	var iaid [4]byte
+	copy(iaid[:], hwaddr[l-4:l])
+	modifiers = append([]Modifier{WithIAID(iaid)}, modifiers...)
```

Comment:
I am worried that if we append a modifier, it will always be applied after the ones passed by parameter. How about directly applying the modifier here?

Reply:
the modifier here is prepended, not appended, so it will always be applied first (and can be overridden)

Reply:
Oh, right, thanks for the clarification.